src/components/request/RequestForm.tsx

```diff
@@ -14,22 +14,44 @@ import {
   RequestInputButtonBlock,
   RequestFormButton,
   BondLogo,
+  ProposerAddress,
+  InputError,
 } from "./Request.styled";
 import { Duration } from "luxon";
 import calculateTimeRemaining from "helpers/calculateTimeRemaining";
 import useClient from "hooks/useOracleClient";
 import useConnection from "hooks/useConnection";
 import useReader from "hooks/useOracleReader";
+import { ChainId, CHAINS } from "constants/blockchain";
+import { ethers } from "ethers";
+import { prettyFormatNumber } from "helpers/format";
 
 const TEN_HOURS_IN_MILLSECONDS = 60 * 60 * 10 * 1000;
 const TWENTY_FOUR_HOURS_IN_MILLISECONDS = 60 * 60 * 24 * 1000;
 const RequestForm: FC = () => {
   const [currentTime, setCurrentTime] = useState(0);
   const [value, setValue] = useState("");
+  const [inputError, setInputError] = useState("");
   const inputOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value);
+    const v = event.target.value;
+    setValue(v);
+    setInputError("");
+    checkForInputError(v);
   };
-  const { flags, client, state } = useClient();
+  const checkForInputError = (v: string) => {
+    if (isNaN(Number(v))) {
+      return setInputError("Must be a valid number.");
+    }
+    if (v.includes(".") && read().collateralProps().decimals) {
```

Comment:
i just rememberd that price proposals are always in 18 decimal regardless of collateral type, so i think this becomes a lot easier, you dont need to import decimals

Reply:
Sounds good, removed.